### PR TITLE
remove unused Functions D to H

### PR DESCRIPTION
**Pull request by CasparKielwein**
Removes unused code in directories starting with Letters D to H, as described in #1955 .

Pretty much analogue to #1952

### CODE REVIEW

File: code/debugconsole/console.h [hunk 1]
```diff
@@ -119,27 +119,6 @@ class debug_command;
...
 	}
 
- /**
-  * @def Shortcut for single-variable setters/monitors
-  *
-  * @param [in] function_name
-  * @param [in] float_variable
-  * @param [in] short_help
-  */
- #define DCF_FLOAT(function_name, float_variable, short_help)	\
```
Comment (asarium) [inline code/debugconsole/console.h]: This should not be removed since it might be useful later.

File: code/debugconsole/console.h [hunk 2]
```diff
@@ -164,27 +143,6 @@ class debug_command;
...
 	}
 
- /**
-  * @def Shortcut for single-variable setters/monitors
-  *
-  * @param [in] function_name
-  * @param [in] int_variable
-  * @param [in] short_help
-  */
- #define DCF_INT(function_name, int_variable, short_help)	\
```
Comment (asarium) [inline code/debugconsole/console.h]: This should not be removed since it might be useful later.

File: code/debugconsole/consoleparse.cpp [hunk 3]
```diff
@@ -456,105 +450,6 @@ bool dc_maybe_stuff_string_white(SCP_string &str)
 	}
 }
 
-void dc_required_string(char *pstr)
```
Comment (asarium) [inline code/debugconsole/consoleparse.cpp]: These are also library functions so they should not be removed.

File: code/debugconsole/console.h (hunk 1 above)
Comment (CasparKielwein) [reply code/debugconsole/console.h]: done

File: code/debugconsole/consoleparse.cpp (hunk 3 above)
Comment (CasparKielwein) [reply code/debugconsole/consoleparse.cpp]: done

**Comment by Goober5000** [comment]
Aside from what @asarium mentioned, I have no objection.

It's interesting from a historical standpoint to see `hud_target_closest_locked_missile`.

**Pull request merged**